bee-message/src/output/alias.rs

```diff
@@ -38,26 +38,15 @@ pub struct AliasOutputBuilder {
 
 impl AliasOutputBuilder {
     ///
-    pub fn new(
-        amount: u64,
-        alias_id: AliasId,
-        state_controller: Address,
-        governor: Address,
-    ) -> Result<AliasOutputBuilder, Error> {
-        validate_controller(&state_controller, &alias_id)?;
-        validate_controller(&governor, &alias_id)?;
-
+    pub fn new(amount: u64, alias_id: AliasId) -> Result<AliasOutputBuilder, Error> {
```

Comment:
> the user was not able to create an invalid object

It's still not possible, it's all part of the builder so you never have an invalid object. The role of a builder is that it can at any point be in an invalid state until the very end when it's converted to a valid object. I conceptually also preferred the old version but it's too confusing because unlock conditions would come from two different places (and that would be the same if we put the params in finish).

I tried many different options but every times it was way more complex than it should have.